File: kotlin/difference-of-squares/src/main/kotlin/DifferenceOfSquares.kt
class Squares(private val tot: Int) {

    private fun Int.sq() = this * this


    fun sumOfSquares(): Int {
        return (0..tot).map { it.sq() }.sum()
    }

    fun squareOfSum(): Int {

        return (0..tot).sum().sq()
    }

    fun difference(): Int {
        return squareOfSum() - sumOfSquares()
    }
}

File: ruby/difference-of-squares/difference_of_squares.rb
=begin
Write your code for the 'Difference Of Squares' exercise in this file. Make the tests in
`difference_of_squares_test.rb` pass.

To get started with TDD, see the `README.md` file in your
`ruby/difference-of-squares` directory.
=end

class Squares

  def initialize (num)
    @num = num
  end

  def square_of_sum
    (1..@num).sum ** 2
  end

  def sum_of_squares
    (1..@num).map { _1 ** 2 }.sum
  end

  def difference
    square_of_sum - sum_of_squares
  end
end
